src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const makeSum = (a) => {
+    if (typeof a === 'number') {
+      sum += a;
+
+      return makeSum;
+    }
```

Comment:
The function should return 'makeSum' regardless of the argument type. Consider simplifying the logic by removing the type check and always returning 'makeSum' if an argument is provided.